### PR TITLE
Detect gzip encoding in Http

### PR DESCRIPTION
**Pull request by szepeviktor**
As seen on TV
https://github.com/crwlrsoft/crawler/blob/a064aeb05d6dcea099cac245190103045efc7606/src/Cache/FileCache.php#L236

### CODE REVIEW

**Comment by szepeviktor** [comment]
After thinking about it `$contents = $message->getBody()->getContents();` should be enough.

- we never want a compressed body
- getContents should detect gzipped-ness, e.g. `Content-Encoding: gzip` too
- getContents should decide compressed contents
- the zlib PHP extension should be required by Composer

**Comment by szepeviktor** [comment]
... just use `symfony/http-client`, it is battle-tested

File: src/Steps/Loading/Http.php [hunk 1]
```diff
@@ -100,7 +101,9 @@ public static function getBodyString(MessageInterface|RespondedRequest $message)
 
             set_error_handler($previousHandler);
 
-            $contents = $decoded === false ? $contents : $decoded;
```
Comment (szepeviktor) [inline src/Steps/Loading/Http.php:103]: This put contents into contents on failure 🥴

**Comment by otsch** [comment]
Thanks!

**Comment by szepeviktor** [comment]
Glad to contribute.

**Pull request merged**